### PR TITLE
Added NumberGenerated functionality and testcases

### PR DESCRIPTION
**Pull request by AbhayMaheshwari123**

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -1,9 +1,16 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
-import App from "./App";
+import NumberGenerator from "./components/NumberGenerator";
```
Comment (nik-1207) [inline src/App.test.tsx]: Number generator test cases should be in numberGenerator.test.tsx

File: src/components/NumberGenerator.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+const NumberGenerator = () => {
+  const [state, setState] = useState<State>({
+    prevNumber: null,
+    currentNumber: 0,
+    pickedNumbers: [],
+  });
+
+  const generateNumber = useCallback(() => {
+    const randomNumber = random(1, 90);
```
Comment (nik-1207) [inline src/components/NumberGenerator.tsx]: constant values should be in a config file

File: tsconfig.json [hunk 3]
```diff
@@ -20,12 +16,6 @@
...
-    "src"
-  ],
-  "exclude": [
-    "node_modules",
-    "coverage",
-    "build"
-  ] // *** The files to not type check ***
-}
\ No newline at end of file
+  "include": ["src"],
```
Comment (nik-1207) [inline tsconfig.json:19]: UNNECESSARY

File: tsconfig.json [hunk 4]
```diff
@@ -20,12 +16,6 @@
...
+  "exclude": ["node_modules", "coverage", "build"] // *** The files to not type check ***
```
Comment (nik-1207) [inline tsconfig.json:20]: UNNECESSARY

File: tsconfig.json [hunk 5]
```diff
@@ -20,12 +16,6 @@
...
+}
```
Comment (nik-1207) [inline tsconfig.json:21]: UNNECESSARY

File: tsconfig.json [hunk 6]
```diff
@@ -1,11 +1,7 @@
 {
   "compilerOptions": {
     "target": "es5",
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "esnext"
-    ],
+    "lib": ["dom", "dom.iterable", "esnext"],
```
Comment (nik-1207) [inline tsconfig.json:4]: UNNECESSARY

File: .vscode/settings.json [hunk 7]
```diff
@@ -1,9 +1,9 @@
 {
```
Comment (nik-1207) [inline .vscode/settings.json:1]: UNNECESSARY

**Review by nik-1207 — COMMENTED** [review]
SEE ONCE

File: src/App.test.tsx (hunk 1 above)
Comment (AbhayMaheshwari123) [reply src/App.test.tsx]: Resolved

File: src/components/NumberGenerator.tsx (hunk 2 above)
Comment (AbhayMaheshwari123) [reply src/components/NumberGenerator.tsx]: resolved

File: tsconfig.json (hunk 3 above)
Comment (AbhayMaheshwari123) [reply tsconfig.json:19]: prettier did this

File: tsconfig.json (hunk 4 above)
Comment (AbhayMaheshwari123) [reply tsconfig.json:20]: prettier did this

File: tsconfig.json (hunk 5 above)
Comment (AbhayMaheshwari123) [reply tsconfig.json:21]: prettier did this

File: tsconfig.json (hunk 6 above)
Comment (AbhayMaheshwari123) [reply tsconfig.json:4]: prettier did this

File: .vscode/settings.json (hunk 7 above)
Comment (AbhayMaheshwari123) [reply .vscode/settings.json:1]: prettier did this

File: src/util/utility.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+const startValue:number=1;
```
Comment (nik-1207) [inline src/util/utility.ts]: SEE NAMING CONVENTION FOR CONSTANT VALUES

File: src/util/utility.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+const endValue:number=90;
```
Comment (nik-1207) [inline src/util/utility.ts]: same here see naming convention for constant value

File: src/util/utility.ts (hunk 8 above)
Comment (AbhayMaheshwari123) [reply src/util/utility.ts]: Resolved

File: src/util/utility.ts (hunk 9 above)
Comment (AbhayMaheshwari123) [reply src/util/utility.ts]: Resolved

File: src/__tests__/NumberGenerator.test.tsx [hunk 10]
```diff
@@ -14,3 +14,24 @@ describe("NumberGenerator ", () => {
...
+    render(<NumberGenerator/>);
+    const btn=screen.getByTestId(/btn/i);
+    expect(btn).toHaveTextContent("Generate Number");
+  });
+  it("should current number generated on button trigger",()=>{
+    const {getByTestId}= render(<NumberGenerator />);
+    fireEvent.click(getByTestId('btn'));
+    expect(getByTestId('btn').innerHTML).not.toBeNull();
+  });
+});
```
Comment (nik-1207) [inline src/__tests__/NumberGenerator.test.tsx]: empty line after  should be there

File: src/components/NumberGenerator.tsx [hunk 11]
```diff
@@ -8,7 +8,7 @@ interface State {
   pickedNumbers: number[];
 }
 
-const NumberGenerator = () => {
```
Comment (nik-1207) [inline src/components/NumberGenerator.tsx]: camel casing

File: src/components/NumberGenerator.tsx [hunk 12]
```diff
@@ -8,7 +8,7 @@ interface State {
...
+const NumberGenerator = ({onClick}:any) => {
```
Comment (nik-1207) [inline src/components/NumberGenerator.tsx]: specify type for functional component

File: src/components/NumberGenerator.tsx [hunk 13]
```diff
@@ -8,7 +8,7 @@ interface State {
...
   const [state, setState] = useState<State>({
```
Comment (nik-1207) [inline src/components/NumberGenerator.tsx]: change name

**Pull request merged**